lib/puppet/functions/provision/get_gcp_profile.rb

```
@@ -0,0 +1,25 @@
+Puppet::Functions.create_function(:'provision::get_gcp_profile') do
```

We prefer using Puppet-language functions. Also, a function should not probably raise an exception - it's up to the caller to handle unexpected results from the function.

---

Done.